internal/store/simple_ttl_store.go

```diff
@@ -0,0 +1,61 @@
+package store
+
+import (
+	"fmt"
+	"time"
+)
+
+// SimpleTTLStore TODO: Thread safety!
+type SimpleTTLStore struct {
+	store    Store
+	ttlOfKey map[string]time.Time
+}
+
+func NewSimpleTTLStore(store Store) *SimpleTTLStore {
+	return &SimpleTTLStore{
+		store:    store,
+		ttlOfKey: make(map[string]time.Time),
+	}
+}
+
+func (s *SimpleTTLStore) SetValue(key string, value string, ttl time.Duration) {
+	s.store.SetValue(key, value)
+	s.SetTTL(key, ttl)
+}
+
+// Does not reset the TTL!
+func (s *SimpleTTLStore) GetValue(key string) (value string, error error) {
+	if !s.isValid(key) {
+		return "", fmt.Errorf("invalid key (too old or not stored)")
+	}
+
+	return s.store.GetValue(key)
+}
+
+func (s *SimpleTTLStore) SetTTL(key string, ttl time.Duration) {
+	s.ttlOfKey[key] = time.Now().Add(ttl)
+}
+
+func (s *SimpleTTLStore) GetTTL(key string) (time.Duration, error) {
```

Comment:
Can this function return an error? By your comment, it should return `0`

Reply:
this implementation yes, but maybe other implementations of this interface do this differently